### PR TITLE
feat(GaussDB): add params to guassdb mysql proxy resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add params to guassdb mysql proxy resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add params to guassdb mysql proxy resource
```

## PR Checklist

* [ ] Tests added/passed.

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccGaussDBMySQLProxy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussDBMySQLProxy_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBMySQLProxy_basic
=== PAUSE TestAccGaussDBMySQLProxy_basic
=== CONT  TestAccGaussDBMySQLProxy_basic
--- PASS: TestAccGaussDBMySQLProxy_basic (4075.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   4075.263s

make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccGaussMysqlEipAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussMysqlEipAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussMysqlEipAssociate_basic
=== PAUSE TestAccGaussMysqlEipAssociate_basic
=== CONT  TestAccGaussMysqlEipAssociate_basic
--- PASS: TestAccGaussMysqlEipAssociate_basic (1450.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1450.292s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
   
不存在

    ab. Related resources (parent resources) not found
    
instance不在了

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    
delete

### CODE REVIEW

File: docs/resources/gaussdb_mysql_proxy.md [hunk 1]
```diff
@@ -30,34 +29,126 @@ The following arguments are supported:
...
+  + **0**: weighted load balancing.
+  + **1**: load balancing (The primary node does not process read requests).
+  + **2**: load balancing (The primary node processes read requests).
+
+* `subnet_id` - (Optional, String, ForceNew) Specifies the network ID of a subnet. Changing this creates a new resource.
+
+* `master_node_weight` - (Optional, List) Specifies the read weight of the master node.
+  The [master_node_weight](#node_weight_struct) structure is documented below.
+
+* `readonly_nodes_weight` - (Optional, List) Specifies the read weight of the read node.
```
Comment (saf3dfsa) [inline docs/resources/gaussdb_mysql_proxy.md]:
```suggestion
* `readonly_nodes_weight` - (Optional, List) Specifies the read weight of the read-only node.
```

File: docs/resources/gaussdb_mysql_proxy.md [hunk 2]
```diff
@@ -30,34 +29,126 @@ The following arguments are supported:
...
+  The [readonly_nodes_weight](#node_weight_struct) structure is documented below.
+
+* `new_node_auto_add_status` - (Optional, String) Specifies whether new nodes are automatically associate with proxy
+  instances. Value options:
+  + **ON**: New nodes are automatically associate with proxy instances.
+  + **OFF**: New nodes are not automatically associate with proxy instances.
+
+  --> **NOTE:** To configure this parameter, contact customer service.
```
Comment (saf3dfsa) [inline docs/resources/gaussdb_mysql_proxy.md]:
```suggestion
  -> To configure this parameter, contact customer service.
```

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy_test.go [hunk 3]
```diff
@@ -1,31 +1,72 @@
...
-	client, err := conf.GaussdbV3Client(acceptance.HW_REGION_NAME)
+func getResourceProxy(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	var (
+		httpUrl = "v3/{project_id}/instances/{instance_id}/proxies"
+		product = "gaussdb"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating GaussDB Client: %s", err)
```
Comment (saf3dfsa) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating GaussDB client: %s", err)
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 4]
```diff
@@ -102,45 +407,489 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+		return fmt.Errorf("error updating GaussDB MySQL port: %s", err)
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	jobId := utils.PathSearch("job_id", updateRespBody, nil)
+	if jobId == "" {
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go]: If jobId is nil, `jobId == ""` will return false, and then assert it is a string will cause error.

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 5]
```diff
@@ -102,45 +407,489 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+		return fmt.Errorf("error updating GaussDB MySQL nodes weight: %s", err)
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	jobId := utils.PathSearch("job_id", updateRespBody, nil)
+	if jobId == "" {
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go]: If jobId is nil, `jobId == ""` will return false, and then assert it is a string will cause error.

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 6]
```diff
@@ -102,45 +407,489 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+
+	return nil
+}
+
+func buildUpdateGaussDBMySQLFlavorBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"flavor_ref": d.Get("flavor"),
+	}
+	return bodyParams
+}
```
Comment (saf3dfsa) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go:505]: It's no need to use d as function param when only use one of the values. Please check the same problems in those functions: buildEnlargeGaussDBMySQLProxyNumberBodyParams, buildUpdateGaussDBMySQLProxyNameBodyParams, buildUpdateGaussDBMySQLPortBodyParams, buildDeleteGaussDBMySQLProxyBodyParams

```suggestion
func buildUpdateGaussDBMySQLFlavorBodyParams(flavor string) map[string]interface{} {
	bodyParams := map[string]interface{}{
		"flavor_ref": flavor,
	}
	return bodyParams
}
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 7]
```diff
@@ -102,45 +407,488 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+	updatePath = strings.ReplaceAll(updatePath, "{proxy_id}", d.Id())
+
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	updateOpt.JSONBody = utils.RemoveNil(buildUpdateGaussDBMySQLFlavorBodyParams(d))
+
+	updateResp, err := client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return fmt.Errorf("error updating GaussDB MySQL flavor: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go]:
```suggestion
		return fmt.Errorf("error updating GaussDB MySQL proxy flavor: %s", err)
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 8]
```diff
@@ -102,45 +407,488 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	result := utils.PathSearch("result", updateRespBody, "").(string)
+	if result != "success" {
+		return fmt.Errorf("error updating GaussDB MySQL new node auto add: result is: %s", result)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go]:
```suggestion
		return fmt.Errorf("error updating GaussDB MySQL proxy  new node auto add: result is: %s", result)
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go [hunk 9]
```diff
@@ -102,45 +407,488 @@ func resourceGaussDBProxyUpdate(ctx context.Context, d *schema.ResourceData, met
...
+	}
+
+	updateRespBody, err := utils.FlattenResponse(updateResp)
+	if err != nil {
+		return err
+	}
+
+	jobId := utils.PathSearch("job_id", updateRespBody, nil)
+	if jobId == nil {
+		return fmt.Errorf("error updating GaussDB MySQL nodes weight: job_id is not found in API response")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_proxy.go]:
```suggestion
		return fmt.Errorf("error updating GaussDB MySQL proxy nodes weight: job_id is not found in API response")
```

**Pull request merged**